mcs/build/gensources.sh

```diff
@@ -1,4 +1,4 @@
-#! /bin/sh
+#!/usr/bin/env bash
```

Comment:
Why is this needed

Reply:
I use some bash functions like [[ - will fail on sh. other .sh files also use it https://github.com/mono/mono/blob/master/mcs/tools/removecomments.sh

Reply:
right but this way it'll look up for anything named `bash` not necessary bash shell

Reply:
some Unix systems don't have bash in /bin so using /usr/bin/env is usually considered more portable.

I do wonder though whether we shouldn't stick to /bin/sh instead and removing the bashisms for such a core script (afaik all the other scripts that use bash aren't required for the build).

Reply:
@akoeplinger  I don't agree this is better, it means that the script could behave differently depending on who runs it.

Reply:
@marek-safar @akoeplinger btw, all .net core build scripts also use "/usr/bin/env bash". I actually changed sh to bash because of linux build machine - macOS and windows(cygwin) work with sh as a first line just fine (as well as my local linux machine). I'll try to rewrite it to sh if it's possible.